src/sdb.c

```diff
@@ -74,7 +74,7 @@ SDB_API Sdb* sdb_new (const char *path, const char *name, int lock) {
 		goto fail;
 	s->ns->free = NULL;
 	if (!s->ns) goto fail;
-	s->ht = ht_new ((SdbListFree)sdb_kv_free);
+	s->ht = ht_new (sdb_hash, strcmp, strdup, strdup, sdb_kv_free);
```

Comment:
eh, it's the cost of haivng it modular, really it also needs another argument for "size" which strlen is used for...